// @sinch/sdk-core/dist/sinch-client.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SinchClient = void 0;
const conversation_1 = require("@sinch/conversation");
const fax_1 = require("@sinch/fax");
const numbers_1 = require("@sinch/numbers");
const sms_1 = require("@sinch/sms");
const verification_1 = require("@sinch/verification");
const voice_1 = require("@sinch/voice");
/** Sinch Client to declare and initialize the supported APIs */
class SinchClient {
    /**
     * Initialize your API Client instance with the provided credentials.
     *
     * @param {SinchClientParameters} params - The object containing the Sinch credentials.
     */
    constructor(params) {
        // Initialize the "Conversation" API
        this.conversation = new conversation_1.ConversationService(params);
        // Initialize the "Fax" API
        this.fax = new fax_1.FaxService(params);
        // Initialize the "Numbers" API
        this.numbers = new numbers_1.NumbersService(params);
        // Initialize the "SMS" API.
        this.sms = new sms_1.SmsService(params);
        // Initialize the "Verification" API
        this.verification = new verification_1.VerificationService(params);
        // Initialize the "Voice" API
        this.voice = new voice_1.VoiceService(params);
    }
}
exports.SinchClient = SinchClient;
